File: CodingTest Lv1/CodingTestPython/collatz.py
def solution(num):
    answer = 0
    while True:
        if(num==1):
            break
        elif (answer == 500):
            answer = -1
            break
        elif (num % 2 == 0):
            num = num // 2
            answer += 1
        elif (num % 2 == 1):
            num = (num * 3) + 1
            answer += 1
    return answer


print(solution(6))
print(solution(16))
print(solution(626331))

File: CodingTest Lv1/CodingTestJava/src/CodingTest/Collatz.java
public class Collatz {
  public static void main(String[] args) {
    Solution sol = new Solution();
    System.out.println(sol.solution(6));
  }
}

class Solution {
  public int solution(int num) {
      int answer = 0;
      long sol_num = (long)num;
      while(true) {
        if(sol_num == 1)
        answer++;
        return answer;
      }
  }
}
